### PR TITLE
feat: add perimeter.cpp

### PR DESCRIPTION
**Pull request by OGscorpion**
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [ x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1897"></a>

### CODE REVIEW

**Comment by OGscorpion** [comment]
Hi, this is my first contribution to an open source project. Please let me know if I made any mistakes.. Thanks!

File: math/perimeter.cpp [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+ * @details The  of a shape is the amount of 2D space it takes up.
+ * All shapes have a formula to get the perimeter of any given shape.
+ * These implementations support multiple return types.
+ *
+ * @author [OGscorpion](https://github.com/OGscorpion)
+ */
+#define _USE_MATH_DEFINES
+#include <cassert>  /// for assert
+#include <cmath>    /// for M_PI definition and pow()
+#include <cmath>
```
Comment (Panquesito7) [inline math/perimeter.cpp:14]: Double `cmath`. 😅
```suggestion
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

**Review by Panquesito7 — DISMISSED** [review]
LGTM 🚀 Thank you for your contribution! 😄👍

**Comment by OGscorpion** [comment]
Thank your for approving my work

File: math/perimeter.cpp [hunk 2]
```diff
@@ -0,0 +1,287 @@
+/**
+ * @file
+ * @brief Implementations for the
+ * [perimeter](https://en.wikipedia.org/wiki/Perimeter) of various shapes
+ * @details The  of a shape is the amount of 2D space it takes up.
+ * All shapes have a formula to get the perimeter of any given shape.
```
Comment (ayaankhan98) [inline math/perimeter.cpp:6]:
```suggestion
 * All shapes have a formula for their perimeter.
```

File: math/perimeter.cpp [hunk 3]
```diff
@@ -0,0 +1,287 @@
+/**
+ * @file
+ * @brief Implementations for the
+ * [perimeter](https://en.wikipedia.org/wiki/Perimeter) of various shapes
+ * @details The  of a shape is the amount of 2D space it takes up.
```
[the remark this answers is not in the document]
Comment (OGscorpion) [reply math/perimeter.cpp:5]: thanks for pointing out

**Review by ayaankhan98 — APPROVED** [review]

**Review by Panquesito7 — APPROVED** [review]

**Review by mishraabhinn — APPROVED** [review]

**Pull request merged**